.github/workflows/build_package.yaml

```diff
@@ -0,0 +1,68 @@
+name: Build pip package
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@main
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
+
+      - uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      # Building the pwheel.
+      # Pypi doesn't have channels like conda, so building the package 
+      # under the name "cpg-hail" to avoid clashes with the official package.
+      - name: Build wheel
+        run: |
+          sed -i 's/name="hail",/name="cpg-hail",/' hail/python/setup.py
+          make -C hail python-version-info wheel
+
+      - name: Install
+        run: |
+          version=$(cat hail/python/hail/hail_pip_version)
+          pip install hail/build/deploy/dist/cpg_hail-$version-py3-none-any.whl
+
+      - name: Test hailctl
+        run: hailctl version
+
+      - name: Test hailtop import
+        run: python -c "import hailtop.batch"
+
+      - name: Test version
+        run: test "$(hailctl version)" = "$(cat hail/python/hail/hail_version)"
+  
+      - name: Publish the wheel to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        if: matrix.os == 'ubuntu-latest' && github.event_name == 'push'
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN }}
+          packages_dir: hail/build/deploy/dist/
+          skip_existing: true
+
+      - name: Redeploy the analysis-server
+        # Only run this step once - that's why limiting to the ubuntu run:
+        if: matrix.os == 'ubuntu-latest' && github.event_name == 'push'
+        run: |
+          pip_version=$(cat hail/python/hail/hail_pip_version)
+          curl --fail --silent --show-error -X POST \
+            -H "Authorization: token ${{ secrets.ANALYSIS_SERVER_GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/populationgenomics/analysis-runner/actions/workflows/6364059/dispatches \
+            -d '{"ref": "main", "inputs": {"hail_version": "$pip_version"}}'
```

Comment:
Does the $pip_version resolve correctly, the dictionary is wrapped in single quotes which I didn't think would work.

Reply:
Could replicate the condarise workflow by putting the version into the GITHUB_ENV, then using the GitHub substitution which doesn't rely on the bash substitution:

```
# previous step
echo "version=$VERSION" >> $GITHUB_ENV

# curl [...]
-d '{"ref": "main", "inputs": {"hail_version": "${{ env.version }}"}}'

Reply:
Good catch!

Reply:
Fixed